openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1920,6 +1920,62 @@ The following is a non-normative example of a Credential Request with Credential
 
 The value of the `credential` claim in the Credential Response MUST be a string that is the base64url-encoded representation of the issued Credential.
 
+## IETF SD-JWT VC
+
+This section defines a Credential Format Profile for Credentials complying with [@!I-D.ietf-oauth-sd-jwt-vc].
+
+### Format Identifier
+
+The Credential format identifier is `vc+sd-jwt`.
+
+### Credential Issuer Metadata {#server_metadata_vc_sd-jwt}
+
+The following additional Credential Issuer metadata parameters are defined for this Credential format to be added to the `credentials_supported` parameter in addition to those defined in (#credential-issuer-parameters).
+
+* `credential_definition`: REQUIRED. Object containing the detailed description of the Credential type. It consists of the following sub elements:
+    * `vct`: REQUIRED. String designating the type of a Credential as defined in [@!I-D.ietf-oauth-sd-jwt-vc].
+    * `claims`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:
+        * `mandatory`: OPTIONAL. Boolean which when set to `true` indicates the claim MUST be present in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.
+        * `value_type`: OPTIONAL. String value determining type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+        * `display`: OPTIONAL. Array of objects, where each object contains display properties of a certain claim in the Credential for a certain language. Below is a non-exhaustive list of valid parameters that MAY be included:
+            * `name`: OPTIONAL. String value of a display name for the claim.
+            * `locale`: OPTIONAL. String value that identifies language of this object represented as language tag values defined in BCP47 [@!RFC5646]. There MUST be only one object for each language identifier.
+* `order`: OPTIONAL. An array of the claim name values that lists them in the order they should be displayed by the Wallet.
```

Comment:
```suggestion
* `vct`: REQUIRED. String designating the type of a Credential as defined in [@!I-D.ietf-oauth-sd-jwt-vc].
* `claims`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:
    * `mandatory`: OPTIONAL. Boolean which when set to `true` indicates the claim MUST be present in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.
    * `value_type`: OPTIONAL. String value determining type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
    * `display`: OPTIONAL. Array of objects, where each object contains display properties of a certain claim in the Credential for a certain language. Below is a non-exhaustive list of valid parameters that MAY be included:
        * `name`: OPTIONAL. String value of a display name for the claim.
        * `locale`: OPTIONAL. String value that identifies language of this object represented as language tag values defined in BCP47 [@!RFC5646]. There MUST be only one object for each language identifier.
* `order`: OPTIONAL. An array of the claim name values that lists them in the order they should be displayed by the Wallet.
```
```

Reply:
Since the proposal is on the table to remove credential_definition I would like to raise another question: why do we treat `vct` special and do not put it underneath `claims`? I think it is desirable to have the metadata be as close as possible to the actual credential structure and `vct` will be just another claim in the SD-JWT VC. 

I would assume we do so because `vct` plays a special role in the request of a credential and `claims` is meant to contain actual user claims, that's why we have the display stuff there. If that's the case, I think we should designate that container as `user_claims`.

Reply:
I am confused what is your suggestion in this comment, Torsten. keep `vct` and `claims` as parameters at the same level, but rename `claims` parameter to `user_claims`?

Reply:
Well, I guess we can agree that neither vct nor iat or exp or iss should turn up here. How do you want to convey that?
At least the text must be clear on that.

Reply:
Would `custom_claims` be a good name for the additional _usage-specific_ claims, that aren't defined by the base JWT spec or the SD-JWT spec.

Reply:
discussed in the DCP WG call - agreed not to rename the parameter name and clarify that the claims are about the subject and not the JWT itself. @Sakurann or @awoie to make a suggestion. good to merge after that.